pkg/sql/jobs/registry.go

```diff
@@ -54,3 +94,202 @@ func (r *Registry) LoadJob(ctx context.Context, jobID int64) (*Job, error) {
 	}
 	return j, nil
 }
+
+// DefaultCancelInterval is a reasonable interval at which to poll this node
+// for liveness failures and cancel running jobs.
+const DefaultCancelInterval = base.DefaultHeartbeatInterval
+
+// DefaultAdoptInterval is a reasonable interval at which to poll system.jobs
+// for jobs with expired leases.
+//
+// DefaultAdoptInterval is mutable for testing. NB: Updates to this value after
+// Registry.Start has been called will not have any effect.
+var DefaultAdoptInterval = 30 * time.Second
+
+// Start polls the current node for liveness failures and cancels all registered
+// jobs if it observes a failure.
+func (r *Registry) Start(
```

Comment:
I'd recommend getting the node's epoch once before waiting out the `cancelInterval`, which I guess would mean just running `maybeCancelJobs` once up front.